GitUI/GitUICommands.cs

```diff
@@ -1690,6 +1694,37 @@ public bool StartFileEditorDialog(string? filename, bool showWarning = false)
             return formEditor.ShowDialog() != DialogResult.Cancel;
         }
 
+        public void StartLocalFileEditorDialog(string? filename)
+        {
+            if (string.IsNullOrWhiteSpace(AppSettings.LocalFileEditor))
+            {
+                StartFileEditorDialog(filename);
+                return;
+            }
+
+            string cmd = AppSettings.LocalFileEditor;
+            Match m = SingleArgRegex.Match(cmd);
+            if (!m.Success)
+            {
+                return;
+            }
+
+            StringBuilder args = new StringBuilder();
+            if (m.Length < cmd.Length)
+            {
+                if (m.Index + m.Length < cmd.Length)
+                {
+                    args.Append(cmd.Substring(m.Index + m.Length).TrimStart(_whiteSpaceChars));
+                    args.Append(" ");
+                }
+
+                cmd = cmd.Substring(m.Index, m.Length);
+            }
```

Comment:
Why is all this needed?

Reply:
Because file names can contain spaces, the chosen editor is allowed to have command line option (e.g. `-multiInst` on the default Notepad++ configuration), and the system needs a clear separate arguments from executable in order for it to work.

Unfortunately, there's no piece of code that does this elsewhere, because calls to the other editor are done from git, which means that the code that basically does the above is in git's codebase.

An alternative would be to run the command using `cmd /c`, but that would force-open a console window, and also kill Linux/Mono compatibility.

Reply:
> Because file names can contain spaces...

This shouldn't be a problem. We handle files with spaces in various situations. Just call `Quote(...)` method.

> ...the chosen editor is allowed to have command line option (e.g. `-multiInst` on the default Notepad++ configuration), 

Why is this a problem?

> ...and the system needs a clear separate arguments from executable in order for it to work.

Why?

Reply:
> also kill Linux/Mono compatibility.

We don't maintain the compatibility since v3.

Reply:
Let me try to explain it as simple as possible. Running:
```
new Executable("Notepad++ -multiInst -nosession").Start("MyFile.cs", useShellExecute: true, throwOnErrorExit: false);
```
will fail with an exception. Running
```
new Executable("Notepad++").Start("-multiInst -nosession MyFile.cs", useShellExecute: true, throwOnErrorExit: false);
```
will work as expected.

The code transforms the prior into the latter. the `Quote` extension cannot do that.